torch/csrc/jit/codegen/cuda/test/test_gpu3.cpp

```diff
@@ -1722,6 +1722,64 @@ TEST_F(NVFuserTest, FusionIndexHoist2_CUDA) {
   testValidate(&fusion, cg_outputs, {t0, t1}, {ref}, __LINE__, __FILE__);
 }
 
+TEST_F(NVFuserTest, FusionIndexHoist3_CUDA) {
+  if (isOptionDisabled(DisableOption::IndexHoist)) {
+    GTEST_SKIP() << "Index hoisting disabled";
+  }
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  auto input = makeContigTensor(2);
+  fusion->addInput(input);
+  auto sin_input = sin(input);
+  auto numel = mul(input->axis(0)->extent(), input->axis(1)->extent());
+  auto output = add(sin_input, numel);
+  fusion->addOutput(output);
+
+  for (auto tv : {output, sin_input}) {
+    tv->merge(0);
+    tv->split(0, 256);
+    tv->axis(0)->parallelize(ParallelType::BIDx);
+    tv->axis(1)->parallelize(ParallelType::TIDx);
+  }
+  inlineMost();
+
+  const auto options = at::TensorOptions().dtype(kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::arange(10000, options).view({100, 100});
+  at::Tensor t1 = t0.sin() + 10000;
+
+  FusionExecutor fe;
+  fe.compileFusion(fusion.get(), {t0});
+  auto cg_outputs = fe.runFusion({t0});
+
+  const std::string expected_kernel = R"(
```

Comment:
I feel that checking the string match for generated kernel is a good idea to test index hoisting. Because when something has changed, we do want to read the new generated code and manually determine if that change makes sense.